pkg/storaged/btrfs/volume.jsx

```diff
@@ -0,0 +1,190 @@
+    const btrfs_volume_card = new_card({
+        title: _("btrfs volume"),
+        next: btrfs_disks_card,
+        page_location: ["btrfs-volume", uuid],
+        page_name: name,
+        page_icon: VolumeIcon,
+        page_size: use[1],
+        component: BtrfsVolumePage,
+        props: { block_devices, uuid, use },
+    });
+
+    const actions = [];
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+    if (!is_mounted) {
+        actions.push({
+            title: _("Mount"),
+            action: () => volume_mount(volume),
+        });
+    }
+
+    const subvolumes_card = new_card({
+        title: _("btrfs subvolumes"),
+        next: btrfs_volume_card,
+        component: BtrfsSubVolumesCard,
+        props: { volume },
+        // TODO: create subvolume
+        actions,
+    });
+
+    const subvolumes_page = new_page(parent, subvolumes_card);
+    make_btrfs_subvolume_pages(subvolumes_page, volume);
+}
+
+const BtrfsVolumePage = ({ card, block_devices, uuid, use }) => {
+    const block_btrfs = client.blocks_fsys_btrfs[block_devices[0].path];
+    const label = block_btrfs.data.label || "-";
+
+    // Changing the label is only supported when the device is not mounted
+    // otherwise we will get btrfs filesystem error ERROR: device /dev/vda5 is
+    // mounted, use mount point This is a libblockdev/udisks limitation as it
+    // only passes the device and not the mountpoint when the device is mounted.
+    // https://github.com/storaged-project/libblockdev/issues/966
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+
+    function rename_dialog() {
+        dialog_open({
+            Title: _("Change label"),
+            Fields: [
+                TextInput("name", _("Name"),
+                          {
+                              validate: name => validate_fsys_label(name, "btrfs"),
+                              value: label
+                          })
+            ],
+            Action: {
+                Title: _("Save"),
+                action: function (vals) {
+                    return block_btrfs.SetLabel(vals.name, {});
+                }
+            }
+        });
+    }
+
+    return (
+        <Stack hasGutter>
+            <StackItem>
+                <StorageCard card={card}>
+                    <CardBody>
+                        <DescriptionList className="pf-m-horizontal-on-sm">
+                            <StorageDescription title={_("Label")}
+                                                value={label}
+                                                action={
+                                                    <StorageLink onClick={rename_dialog}
+                                                               excuse={is_mounted ? _("Btrfs volume is mounted") : null}>
+                                                        {_("edit")}
+                                                    </StorageLink>}
```

Comment:
These 13 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110913-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)